applications/GeoMechanicsApplication/custom_constitutive/constitutive_type.h

```diff
@@ -0,0 +1,30 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#pragma once
+
+#include "containers/flags.h"
+#include "includes/ublas_interface.h"
+
+namespace Kratos
+{
+
+class ConstitutiveType
```

Comment:
Still struggling to get a suitable name here.

Reply:
ConstitutiveLawDimension

A ConstitutiveType for me is more an indication for what material type ( sand, steel, glas etc ) or behaviour ( elastic, plastic, brittle, viscoelastic, viscous etc. ) the constitutive law should be used.

Reply:
I like @WPK4FEM's suggestion of `ConstitutiveLawDimension`. When you change the type name, you may need to change some variable names, too.